index.js

```diff
@@ -50,6 +62,40 @@ function basepath(config, service, opts) {
   return `${config.servicePath}/.serverless/${opts.function || service.service}`;
 }
 
+const readFile = bluebird.promisify(fs.readFile);
+const writeFile = bluebird.promisify(fs.writeFile);
+
+const applyZipExclude = bluebird.coroutine(
+  function*(serverless, opts) {
+    // TODO respect exclude on individual functions
+    const exclude = _.uniq(_.get(serverless.package, "exclude", []));
+
+    if (!_.isEmpty(exclude)) {
+      const data = yield readFile(serverless.service.__cljsArtifact);
+
+      const zip = yield jszip.loadAsync(data);
+
+      const zipObjs = zip.filter(function(relativePath, file) {
```

Comment:
I don't think `zipObjs` deserves to be named, I would prefer if this and the forEach were combined into a single expression:

```js
zip
 .filter((path, file) => !_.some(exclude, pattern => minimatch(path, pattern)))
 .forEach(entry => newZip.file(entry.name, name.nodeStream("nodebuffer")))
```